backend-comparison/src/burnbenchapp/base.rs

```diff
@@ -81,45 +99,94 @@ pub(crate) enum BenchmarkValues {
+            return;
         }
-        Commands::Run(run_args) => {
-            if run_args.backends.is_empty() || run_args.benches.is_empty() {
-                println!("No backends or benchmarks specified. Please select at least one backend and one benchmark.");
-                return;
-            }
-
-            let total_combinations = run_args.backends.len() * run_args.benches.len();
-            println!(
-                "Executing the following benchmark and backend combinations (Total: {}):",
-                total_combinations
-            );
-            for backend in &run_args.backends {
-                for bench in &run_args.benches {
-                    println!("- Benchmark: {}, Backend: {}", bench, backend);
-                }
-            }
-
-            let mut app = App::new();
-            app.init();
-            println!("Running benchmarks...");
-            app.run(&run_args.benches, &run_args.backends);
-            app.cleanup();
-            println!("Cleanup completed. Benchmark run(s) finished.");
+    };
+    println!("🌐 Please visit for following URL in your browser (CTRL+click if your terminal supports it):");
+    println!("\n    {}\n", flow.verification_uri.clone().unwrap());
+    let user_code = flow.user_code.clone().unwrap();
+    println!("👉 And enter code: {}", &user_code);
+    if let Ok(mut clipboard) = Clipboard::new() {
+        if clipboard.set_text(user_code).is_ok() {
+            println!("📋 Code has been successfully copied to clipboard.")
+        };
+    };
+    thread::sleep(FIVE_SECONDS);
+    match flow.poll(20) {
+        Ok(creds) => {
+            save_token(&creds.token);
         }
+        Err(e) => eprint!("Authentication error: {}", e),
+    };
+}
+
+fn command_list() {
+    println!("Available Backends:");
+    for backend in BackendValues::iter() {
+        println!("- {}", backend);
+    }
+    println!("\nAvailable Benchmarks:");
+    for bench in BenchmarkValues::iter() {
+        println!("- {}", bench);
+    }
+}
+
+fn command_run(run_args: RunArgs) {
+    if run_args.share {
+        // Verify if a token is saved
+        let token = get_token_from_cache();
+        if token.is_none() {
+            eprintln!("You need to be authenticated to be able to share benchmark results.");
+            eprintln!("Run the command 'burnbench auth' to authenticate.");
+            return;
+        }
+        // TODO refresh the token when it is expired
+        // Check for the validity of the saved token
+        if !is_token_valid(&token.unwrap()) {
+            eprintln!("Your access token is no longer valid.");
+            eprintln!("Run the command 'burnbench auth' again to get a new token.");
+            return;
+        }
+    }
+    if run_args.backends.is_empty() || run_args.benches.is_empty() {
+        println!("No backends or benchmarks specified. Please select at least one backend and one benchmark.");
+        return;
+    }
```

Comment:
I think we can set the mininum number of `backends` and `benches` with clap (`num_args(0..)` to `num_args(1..)`)

Reply:
Good point !!

Reply:
Made the fix along with setting the --benches and --backends arguments as required. I also fixed the documentation for these arguments that was wrong. 🙂